hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/net/SocketOutputStream.java

```diff
@@ -219,7 +226,9 @@ public void transferToFully(FileChannel fileCh, long position, int count,
       waitForWritable();
       long wait = System.nanoTime();
 
-      int nTransfered = (int) fileCh.transferTo(position, count, getChannel());
+      int nTransfered = (int) NativeIO.transferTo(fileCh,
+          fileInputStream.getFD(), getChannel(),
+          ((SelChImpl) getChannel()).getFD(), position, count);
```

Comment:
I cannot find an easy way out but can you see if we can remove:
SelChImpl is internal proprietary API and may be removed in a future release